pkg/peerhealth/client_server_test.go

```diff
@@ -111,11 +111,12 @@ var _ = Describe("Checking health using grpc client and server", func() {
 			By("calling isHealthy")
 			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 			defer (cancel)()
-			resp, err := phClient.IsHealthy(ctx, &HealthRequest{
-				NodeName: nodeName,
-			})
-			Expect(err).ToNot(HaveOccurred())
-			Expect(api.HealthCheckResponseCode(resp.Status)).To(Equal(api.Unhealthy))
+			Eventually(func() bool {
+				resp, err := phClient.IsHealthy(ctx, &HealthRequest{
+					NodeName: nodeName,
+				})
+				return err == nil && api.HealthCheckResponseCode(resp.Status) == api.Unhealthy
+			}, time.Millisecond*250, time.Second*5).Should(BeTrue())
```

Comment:
IIUC then you messed up with the order of parameters here. 
The first parameter would be timeout, and the second would be a polling interval https://pkg.go.dev/github.com/onsi/gomega#Eventually.

Using these variables it seems like `Eventually` ran one and wouldn't be retriggered, since `timeout <  polling interval` ( e.g., 250ms < 5s) . But if these variables were good for fixing the flakiness and it is indeed run once (maybe add some debugging prints inside to verify my assumption), then you don't need this new _Eventually_ and you can revert to what we have before.

Reply:
Thanks, good catch